### ISSUE TITLE
DOMException is thrown when a data attribute contains dots

### ISSUE THREAD

**Issue by sebastianfrey** [forsti0506/a11y-sitechecker#533]
When a web page declares data attributes with dots the accessibility checker fails and throws an uncaught DOMException.

Example for such a data attribute:

`<body data-keydown.tabguard="1" />`

I looked around the code base and I think I've found the reason causing the exception:

https://github.com/forsti0506/a11y-sitechecker/blob/0d14a409dd559dfd8843cf9ce4dae43ebf2329ff/lib/utils/unique-selector.ts#L23

As I understand that peace of code correctly, everything after the dot and inclusive the dot is replaced with `class=[${everything-after-dot}]`.

This turns the above example into the following invalid selector:

`body[data-keydown[class='tabguard="1"]']`

In such a case, the query selector should escape the dot with double slashes:

`body[data-keydown\\.tabguard="1"]`

If desired, I am willing to provide a PR.
